lib/policies/jwt/jwt.js

```diff
@@ -0,0 +1,42 @@
+const fs = require('fs');
+const passport = require('passport');
+const JWTStrategy = require('passport-jwt').Strategy;
+const extractors = require('./extractors');
+const services = require('../../services');
+
+module.exports = function (params) {
+  const secret = params.secretFile ? fs.readFileSync(params.secretFile) : params.secret;
+  const extractor = extractors[params.jwtExtractor](params.jwtExtractorField);
+
+  passport.use(new JWTStrategy({
+    secretOrKey: secret,
+    jwtFromRequest: extractor,
+    audience: params.audience,
+    issuer: params.issuer
+  }, (jwtPayload, done) => {
+    if (!jwtPayload || !jwtPayload.sub) {
+      return done(null, false);
```

Comment:
I think we need an option here where we _don't_ validate the user in Express Gateway.  There are plenty of use cases where I just want to verify the integrity of the JWT token and pass it along.  If using an external Identity Provider (like Auth0), it will be a bit of a pain to have to import and synchronize those user records.

Reply:
Hold on sir — we're not validating the user here, as we do not have such information (or probably we have, but we don't know where it's stored). `sub` has the credential ID here, not the user ID.

Reply:
Pardon my slip.  Yes, it's the credential ID.  The point is still the same.  There are plenty of projects that won't use our identity management system at all and only need to verify the token before proxying to a backend.  We should support that use case.

Reply:
The credential ID is not really part of the identity system, it's nothing more than a list of keys the gateway knows that might come.

Creating a new key in the gateway should not really be a big deal. Let's keep in mind that a JWT key is per consumer, not per user: 1 key can handle millions users. I don't see any giant amounts of data to import, in the eventuality of a migration to EG.

If we renounce on this, the gateway would trust any JWT with any content inside, as long the signature is valid, which I believe it's harmful.

Keep in mind, for instance, that JWT tokens cannot be revoked. Is a system gets compromised, the tokens will continue to flow and to be validated correctly as long you do not replace your secret or private key. This could break all the clients.

Having the keyId stored in the gateway as a mandatory parameters will let the user have a way to revoke a consumer from doing requests, if necessary. I believe it's a valuable security feature to have.

I can live with such a flag, but I am not completely sure it's the way to go.

Maybe @altsang has his opinion on that as well?

Reply:
@kevinswiber 
I've implemented that flag with tests as well:
Feature: bd43b46
Tests: b49682e